compose/progress_stream.py

```diff
@@ -27,15 +30,18 @@ def stream_output(output, stream):
                 stream.write("\n")
                 diff = 0
 
-            if is_terminal:
-                # move cursor up `diff` rows
-                stream.write("%c[%dA" % (27, diff))
+            # move cursor up `diff` rows
+            stream.write("%c[%dA" % (27, diff))
+
+            print_output_event(event, stream, is_terminal)
 
-        print_output_event(event, stream, is_terminal)
+            if 'id' in event:
+                # move cursor back down
+                stream.write("%c[%dB" % (27, diff))
 
-        if 'id' in event and is_terminal:
-            # move cursor back down
-            stream.write("%c[%dB" % (27, diff))
+        # print all other events
+        elif not is_progress_event:
+            print_output_event(event, stream, is_terminal)
```

Comment:
To make this a bit easier to read, and remove the need for the second check of `is_progress_event`, could we switch this around and do:

```python

if not is_progress_event:
    print_output_event(event, stream, is_terminal)
    continue

# The rest of the logic below here
...

```